devicetypes/smartthings/ozom-smart-siren.src/ozom-smart-siren.groovy

```diff
@@ -127,32 +139,37 @@ def configure() {
 
 def both() {
 	log.debug "both()"
-	on()
+	state.alarmCmd = ALARM_BOTH
+	def warningDuration = state.maxDuration ? state.maxDuration : DEFAULT_MAX_DURATION
+	state.lastDuration = warningDuration
+
+	zigbee.command(IAS_WD_CLUSTER, COMMAND_IAS_WD_START_WARNING, "17", DataType.pack(warningDuration, DataType.UINT16), "40", "03")
 }
 
 def siren() {
 	log.debug "siren()"
-	on()
-}
 
-def strobe() {
-	log.warn "strobe() is not supported by this device"
+	state.alarmCmd = ALARM_SIREN
+	def warningDuration = state.maxDuration ? state.maxDuration : DEFAULT_MAX_DURATION
+	state.lastDuration = warningDuration
+
+	// start warning, burglar mode, no strobe, siren very high
+	zigbee.command(IAS_WD_CLUSTER, COMMAND_IAS_WD_START_WARNING, "13", DataType.pack(warningDuration, DataType.UINT16), "00", "00")
 }
 
-def on() {
-	log.debug "on()"
+def strobe() {
+	log.debug "strobe()"
 
-	state.alarmOn = true
+	state.alarmCmd = ALARM_STROBE
 	def warningDuration = state.maxDuration ? state.maxDuration : DEFAULT_MAX_DURATION
 	state.lastDuration = warningDuration
 
-	// start warning, burglar mode, no strobe, siren very high
-	zigbee.command(IAS_WD_CLUSTER, COMMAND_IAS_WD_START_WARNING, "13", DataType.pack(warningDuration, DataType.UINT16), "00", "00")
+	zigbee.command(IAS_WD_CLUSTER, COMMAND_IAS_WD_START_WARNING, "04", DataType.pack(warningDuration, DataType.UINT16), "40", "03")
```

Comment:
I know the code had magic numbers before, but it would be better to create functions to return the numbers in order to make the code more self documenting.